### PR TITLE
Merge dev to master

### PR DESCRIPTION
**Pull request by Raycoms**
New schematics
New achievements

Review please @Minecolonies/maintainers

### CODE REVIEW

**Comment by Raycoms** [comment]
We'll have to merge master to dev first it seems, can someone do that?

File: src/main/java/com/minecolonies/items/ModItems.java [hunk 1]
```diff
@@ -9,6 +9,11 @@
     public static final Item scanTool    = new ItemScanTool();
     public static final Item caliper     = new ItemCaliper();
 
+    public static final Item itemAchievementProxySettlement = new ItemAchievementProxy("sizeSettlement");
+    public static final Item itemAchievementProxyTown       = new ItemAchievementProxy("sizeTown");
+    public static final Item itemAchievementProxyCity       = new ItemAchievementProxy("sizeCity");
+    public static final Item itemAchievementProxyMetropolis = new ItemAchievementProxy("sizeMetropolis");
+
     // deactivated for now
     // public static final Item compost    = new ItemCompost();
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/items/ModItems.java]: MAJOR This block of commented-out lines of code should be removed. [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3ACommentedOutCodeLine)

File: src/main/java/com/minecolonies/items/ItemCompost.java [hunk 2]
```diff
@@ -22,6 +23,8 @@
     public ItemCompost()
     {
         super("compost");
+
+        super.setCreativeTab(ModCreativeTabs.MINECOLONIES);
         maxStackSize = 64;
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/items/ItemCompost.java]: MINOR Assign this magic number 64 to a well-named constant, and use the constant instead. [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS109)

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 13 issues
* MAJOR 5 major
* MINOR 8 minor

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [Colony.java#L657](https://github.com/Minecolonies/minecolonies/blob/51425714874d3d46e2074c0d61c22588bdea5bb6/src/main/java/com/minecolonies/colony/Colony.java#L657): Remove this unused method parameter "oldSubscribers". [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1172)
1. MAJOR [Colony.java#L659](https://github.com/Minecolonies/minecolonies/blob/51425714874d3d46e2074c0d61c22588bdea5bb6/src/main/java/com/minecolonies/colony/Colony.java#L659): Add parentheses to make the operator precedence explicit. [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS864)
1. MAJOR [ItemCompost.java#L32](https://github.com/Minecolonies/minecolonies/blob/51425714874d3d46e2074c0d61c22588bdea5bb6/src/main/java/com/minecolonies/items/ItemCompost.java#L32): Add the "@Override" annotation above this method signature [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1161)
1. MAJOR [ItemCompost.java#L43](https://github.com/Minecolonies/minecolonies/blob/51425714874d3d46e2074c0d61c22588bdea5bb6/src/main/java/com/minecolonies/items/ItemCompost.java#L43): Merge this if statement with the enclosing one. [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1066)
1. MINOR [BuildingFarmer.java#L208](https://github.com/Minecolonies/minecolonies/blob/51425714874d3d46e2074c0d61c22588bdea5bb6/src/main/java/com/minecolonies/colony/buildings/BuildingFarmer.java#L208): Define a constant instead of duplicating this literal "com.minecolonies.gui.scarecrow.user" 3 times. [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1192)
1. MINOR [ItemCompost.java#L18](https://github.com/Minecolonies/minecolonies/blob/51425714874d3d46e2074c0d61c22588bdea5bb6/src/main/java/com/minecolonies/items/ItemCompost.java#L18): Document this public class. [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AUndocumentedApi)
1. MINOR [ItemCompost.java#L32](https://github.com/Minecolonies/minecolonies/blob/51425714874d3d46e2074c0d61c22588bdea5bb6/src/main/java/com/minecolonies/items/ItemCompost.java#L32): Document this public method. [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AUndocumentedApi)
1. MINOR [ItemCompost.java#L35](https://github.com/Minecolonies/minecolonies/blob/51425714874d3d46e2074c0d61c22588bdea5bb6/src/main/java/com/minecolonies/items/ItemCompost.java#L35): Local variable 'iblockstate' could be declared final [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3ALocalVariableCouldBeFinal)
1. MINOR [ItemCompost.java#L39](https://github.com/Minecolonies/minecolonies/blob/51425714874d3d46e2074c0d61c22588bdea5bb6/src/main/java/com/minecolonies/items/ItemCompost.java#L39): Local variable 'igrowable' could be declared final [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3ALocalVariableCouldBeFinal)
1. MINOR [ItemScanTool.java#L26](https://github.com/Minecolonies/minecolonies/blob/51425714874d3d46e2074c0d61c22588bdea5bb6/src/main/java/com/minecolonies/items/ItemScanTool.java#L26): Split this 184 characters long line (which is greater than 180 authorized). [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS00103)

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 13 issues
* MAJOR 5 major
* MINOR 8 minor

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [Colony.java#L657](https://github.com/Minecolonies/minecolonies/blob/36d9693ee595f0312fdc8efb5434aab3bce1d39d/src/main/java/com/minecolonies/colony/Colony.java#L657): Remove this unused method parameter "oldSubscribers". [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1172)
1. MAJOR [Colony.java#L659](https://github.com/Minecolonies/minecolonies/blob/36d9693ee595f0312fdc8efb5434aab3bce1d39d/src/main/java/com/minecolonies/colony/Colony.java#L659): Add parentheses to make the operator precedence explicit. [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS864)
1. MAJOR [ItemCompost.java#L32](https://github.com/Minecolonies/minecolonies/blob/36d9693ee595f0312fdc8efb5434aab3bce1d39d/src/main/java/com/minecolonies/items/ItemCompost.java#L32): Add the "@Override" annotation above this method signature [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1161)
1. MAJOR [ItemCompost.java#L43](https://github.com/Minecolonies/minecolonies/blob/36d9693ee595f0312fdc8efb5434aab3bce1d39d/src/main/java/com/minecolonies/items/ItemCompost.java#L43): Merge this if statement with the enclosing one. [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1066)
1. MINOR [BuildingFarmer.java#L208](https://github.com/Minecolonies/minecolonies/blob/36d9693ee595f0312fdc8efb5434aab3bce1d39d/src/main/java/com/minecolonies/colony/buildings/BuildingFarmer.java#L208): Define a constant instead of duplicating this literal "com.minecolonies.gui.scarecrow.user" 3 times. [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1192)
1. MINOR [ItemCompost.java#L18](https://github.com/Minecolonies/minecolonies/blob/36d9693ee595f0312fdc8efb5434aab3bce1d39d/src/main/java/com/minecolonies/items/ItemCompost.java#L18): Document this public class. [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AUndocumentedApi)
1. MINOR [ItemCompost.java#L32](https://github.com/Minecolonies/minecolonies/blob/36d9693ee595f0312fdc8efb5434aab3bce1d39d/src/main/java/com/minecolonies/items/ItemCompost.java#L32): Document this public method. [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AUndocumentedApi)
1. MINOR [ItemCompost.java#L35](https://github.com/Minecolonies/minecolonies/blob/36d9693ee595f0312fdc8efb5434aab3bce1d39d/src/main/java/com/minecolonies/items/ItemCompost.java#L35): Local variable 'iblockstate' could be declared final [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3ALocalVariableCouldBeFinal)
1. MINOR [ItemCompost.java#L39](https://github.com/Minecolonies/minecolonies/blob/36d9693ee595f0312fdc8efb5434aab3bce1d39d/src/main/java/com/minecolonies/items/ItemCompost.java#L39): Local variable 'igrowable' could be declared final [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3ALocalVariableCouldBeFinal)
1. MINOR [ItemScanTool.java#L26](https://github.com/Minecolonies/minecolonies/blob/36d9693ee595f0312fdc8efb5434aab3bce1d39d/src/main/java/com/minecolonies/items/ItemScanTool.java#L26): Split this 184 characters long line (which is greater than 180 authorized). [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS00103)

**Comment by Raycoms** [comment]
Can we please move this @Minecolonies/maintainers

**Comment by Kostronor** [comment]
:shipit:

**Pull request merged**